testing/python/collect.py

```diff
@@ -143,6 +143,22 @@ class test(object):
             "*collected 0*",
         ])
 
+    def test_static_method(self, testdir):
+        testdir.getmodulecol("""
+            class Test(object):
+                @staticmethod
+                def test_something():
+                    pass
+        """)
+        result = testdir.runpytest()
+        if sys.version_info < (2,7):
+            # in 2.6, the code to handle static methods doesn't work
+            result.stdout.fnmatch_lines(["*collected 0*"])
```

Comment:
Please improve matching here:

```python
if sys.version_info < (2, 7):
    result.stdout.fnmatch_lines([
        "*collected 0 items*",
        "*cannot collect static method*",
    ])
else:
    result.stdout.fnmatch_lines([
        "*collected 1*",
        "*1 passed in*",
    ])
```

This way we ensure that we are showing the appropriate warning and that we are collecting and running it in Py27+.